strings-csharp/StringFormattableString/StringFormattableStringTests/FormattableStringsUnitTests.cs

```diff
@@ -0,0 +1,95 @@
+using Microsoft.Extensions.Logging;
+
+namespace StringFormattableStringTests
+{
+    [TestClass]
+    public class FormattableStringsUnitTests
+    {
+        private readonly FormattableStringsMethods _formattableStringMethods = new();
+    
+        [TestMethod]
+        public void GivenAStringObject_WhenStringOperationsApplied_VerifyAccurateResults()
+        {
+            const string studentName = "John";
+            const int studentAge = 30;
+            var sampleString = _formattableStringMethods.StringExample(studentName, studentAge);
+
+            Assert.IsInstanceOfType(sampleString, typeof(string)); 
+            Assert.IsTrue(sampleString.Contains(studentName));
+            Assert.IsTrue(sampleString.Contains(studentAge.ToString()));
+        }
+
+        [TestMethod]
+        public void GivenAFormattableStringObject_WhenFormattableStringOperationsApplied_VerifyAccurateResults()
+        {
+            var studentName = "Sean";
+            var studentAge = 40;
+            var sampleFormattableString = _formattableStringMethods.FormattableStringExample(studentName, studentAge);
+
+            Assert.AreEqual(studentName, sampleFormattableString.GetArgument(0));
+            Assert.AreEqual(studentAge, sampleFormattableString.GetArgument(1));
+            Assert.AreEqual(sampleFormattableString.ArgumentCount, 2);
+            Assert.IsInstanceOfType(sampleFormattableString, typeof(FormattableString));
```

Comment:
Same argument for reordering would apply here. We should try to think about our assertions as a process going from general to more specific, especially in the context of multiple assertions within a test. An assertion is our way of "proving" something and so the assertions should always follow a logical ordering and be related to either what we are:
1. setting up (i.e. if we are testing that a method adds something to an empty list, we could first assert that the list is empty as part of our setup, as a way of "proving" that the test is valid).
2. we are validating, but again in the presence of multiple assertions to move from general to specific.